neffint/fixed_grid_fourier_integral.py

```diff
@@ -0,0 +1,247 @@
+    abs_x_to_n_plus_1 = np.abs(xx) # Starts as x**(0+1)
+    exp_abs_x = np.exp(np.abs(xx))
+
+    # Run convergence loop until phi and psi converged
+    phi_converged = False
+    psi_converged = False
+    for n in range(MAX_PHI_PSI_ITERATIONS+1):
+
+        # Stop loop if converged
+        if phi_converged and psi_converged:
+            break
+
+        # Only calculate phi if not yet converged (to save computation time)
+        if not phi_converged:
+            phi[taylor_mask] += (n+6)/((n+3)*(n+4)) * jx_to_n * inv_factorial
+
+            # Check for convergence
+            min_phi_component = np.min([np.abs(np.real(phi[taylor_mask])), np.abs(np.imag(phi[taylor_mask]))], axis=0)
+            phi_remaining_terms_bound = 2 * abs_x_to_n_plus_1 * exp_abs_x * inv_factorial / ((n+1)*(n+5))
+            phi_converged = np.all(phi_remaining_terms_bound < rel_tolerance * min_phi_component)
+        
+        # Only calculate pis if not yet converged
+        if not psi_converged:
+            psi[taylor_mask] += - 1/((n+3)*(n+4)) * jx_to_n * inv_factorial
+
+            # Check for convergence
+            min_psi_component = np.min([np.abs(np.real(psi[taylor_mask])), np.abs(np.imag(psi[taylor_mask]))], axis=0)
+            psi_remaining_terms_bound = abs_x_to_n_plus_1 * exp_abs_x * inv_factorial / ((n+1)*(n+4)*(n+5))
+            psi_converged = np.all(psi_remaining_terms_bound < rel_tolerance*min_psi_component)
+        
+        # Update variables
+        jx_to_n *= 1j*xx
+        abs_x_to_n_plus_1 *= np.abs(xx)
+        inv_factorial /= n+1
+
+    # If loop finished without converging, raise error
+    if not (phi_converged and psi_converged):
+        raise RuntimeError(f"Phi and Psi calculation failed to converge after {MAX_PHI_PSI_ITERATIONS} iterations")
+    
+    return phi, psi
+
+def fourier_integral_inf_correction(times: np.ndarray, omega_end: float, func_value_end: ArrayLike, func_derivative_end: ArrayLike=0.):
+    """Calculate the asymptotic correction term as omega->inf of a Fourier integral for the given times.
+    Uses a first or second (if func_derivative_end is given) order Taylor expansion around the angular frequency omega_end.
+
+    :param times: A 1D array of length M with the times to compute the Fourier integral for
+    :type times: np.ndarray
+    :param omega_end: A frequency to expand the Taylor series from
+    :type omega_end: float
+    :param func_value_end: An array of shape (X1, X2, ...) containing the values of the function to be transformed evaluated at omega_end
+    :type func_value_end: ArrayLike
+    :param func_derivative_end: An array of shape (X1, X2, ...) containing the derivative of the function to be transformed, evaluated at omega_end, defaults to 0.
+    :type func_derivative_end: ArrayLike, optional
+    :return: The asymptotic terms of the Fourier Integral for all times, given as an array of shape (M, X1, X2, ...)
+    :rtype: np.ndarray
```

Comment:
I did not mention units in the docstring (yet). I think it is probably a good idea to mention it, but not sure if I should just write [s] and [Hz] or if I should write that any coherent time and frequency units work (e.g. ns and GHz should work, so should m and m^-1)

Reply:
I agree including the units you suggest. Might annotate them with a `*` and add a footnote like

    * Any coherent time and frequency units will work.

PS: I see I copied the review comment from the wrong docstring, hence the `2pi` mixup